src/instructlab/model/evaluate.py

```diff
@@ -1,52 +1,145 @@
 # Third Party
 from click_didyoumean import DYMGroup
 from instructlab.eval import (
-    MMLU_Evaluator,
-    MT_Bench_Evaluator,
-    PR_Bench_Evaluator,
-    PR_MMLU_Evaluator,
+    Evaluator,
+    MMLUBranchEvaluator,
+    MMLUEvaluator,
+    MTBenchBranchEvaluator,
+    MTBenchEvaluator,
 )
 import click
 
-benchmark_names_to_classes = frozenset({
-    "mmlu": MMLU_Evaluator,
-    "mt_bench": MT_Bench_Evaluator,
-    "pr_bench": PR_Bench_Evaluator,
-    "pr_mmlu": PR_MMLU_Evaluator,
-})
+BENCHMARK_TO_CLASS_MAP = frozenset(
+    {
+        "mmlu": MMLUEvaluator,
+        "mmlu_branch": MMLUBranchEvaluator,
+        "mt_bench": MTBenchEvaluator,
+        "mt_bench_branch": MTBenchBranchEvaluator,
+    }
+)
+
+
+def get_evaluator(
+    model_name,
+    benchmark,
+    judge_model_name,
+    output_dir,
+    max_workers,
+    taxonomy_path,
+    branch,
+    tasks,
+    few_shots,
+    batch_size,
+    task,
+    sdg_path,
+) -> Evaluator:
+    """takes in arguments from the CLI and uses 'benchmark' to validate other arguments
+    if all needed configuration is present, returns the appropriate Evaluator class for the benchmark
+    otherwise raises an exception for the missing values
+    """
+
+    # ensure skills benchmarks have proper arguments if selected
+    if benchmark in ["mt_bench", "mt_bench_branch"]:
+        required_args = [
+            model_name,
+            judge_model_name,
+            output_dir,
+            max_workers,
+        ]
+        if benchmark == "mt_bench_branch":
+            required_args.append(taxonomy_path, branch)
+        if any(required_args) is None:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_args}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        else:
+            evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+            if benchmark == "mt_bench":
+                return evaluator_class(
+                    model_name, judge_model_name, output_dir, max_workers
+                )
+            else:
+                return evaluator_class(
+                    model_name,
+                    judge_model_name,
+                    taxonomy_path,
+                )
+
+    # ensure knowledge benchmarks have proper arguments if selected
+    if benchmark in ["mmlu", "mmlu_branch"]:
+        required_args = [few_shots, batch_size]
+        if benchmark == "mmlu":
+            required_args.append(tasks)
+        if benchmark == "mmlu_branch":
```

Comment:
elif

Reply:
Any particular reason? I don't see a functional difference

Reply:
I would suggest it makes the code more clear (that it will never hit the second if if the first one succeeds) and it avoid the extra unneeded string compare in those cases.

Reply:
> `elif` would still require the string compare no?

the elif wouldn't compare if the first if was a match